[CI] BytesReadRankFeatureIT testRankFeaturePhaseEmptyDocIdsStillReportsDirectoryMetrics failing

**Build Scans:**
- [elasticsearch-intake #41466 / part1](https://gradle-enterprise.elastic.co/s/lyz2qmfuare6q)
- [elasticsearch-pull-request #153852 / part-1](https://gradle-enterprise.elastic.co/s/oji62omkzh4uk)
- [elasticsearch-pull-request #153640 / part-1](https://gradle-enterprise.elastic.co/s/e6tueb4a7tdoq)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.index.store.BytesReadRankFeatureIT.testRankFeaturePhaseEmptyDocIdsStillReportsDirectoryMetrics" -Dtests.seed=B01B2FF57D819A09 -Dtests.locale=sq-Latn-AL -Dtests.timezone=Europe/Brussels -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.store.BytesReadRankFeatureIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRankFeaturePhaseEmptyDocIdsStillReportsDirectoryMetrics'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: invalid decRef call: already closed
```

**Issue Reasons:**
- [main] 3 failures in test testRankFeaturePhaseEmptyDocIdsStillReportsDirectoryMetrics (1.2% fail rate in 243 executions)
- [main] 2 failures in step part-1 (1.3% fail rate in 156 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.3% fail rate in 155 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -605,9 +605,6 @@ tests:
 - class: org.elasticsearch.upgrades.MultiVersionRepositoryAccessIT
   method: testUpgradeMovesRepoToNewMetaVersion
   issue: https://github.com/elastic/elasticsearch/issues/151391
-- class: org.elasticsearch.index.store.BytesReadRankFeatureIT
-  method: testRankFeaturePhaseEmptyDocIdsStillReportsDirectoryMetrics
-  issue: https://github.com/elastic/elasticsearch/issues/151392
 - class: org.elasticsearch.xpack.stateless.cluster.coordination.TransportConsistentClusterStateReadIT
   method: testWaitsUntilTheNodeRejoinsTheClusterToReturnAValidClusterState
   issue: https://github.com/elastic/elasticsearch/issues/151402

--- a/server/src/internalClusterTest/java/org/elasticsearch/index/store/BytesReadRankFeatureIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/index/store/BytesReadRankFeatureIT.java
@@ -150,13 +150,7 @@ public class BytesReadRankFeatureIT extends ESIntegTestCase {
             );
         } finally {
             if (queryResult != null) {
-                if (queryResult.hasReferences()) {
-                    queryResult.decRef();
-                }
                 searchService.freeReaderContext(queryResult.getContextId());
-            }
-            if (rankResult != null && rankResult.hasReferences()) {
-                rankResult.decRef();
             }
         }
     }

